code/game/objects/machinery/robotic_cradle.dm

```diff
@@ -71,6 +65,8 @@
 	if(!repairing)
 		return
 
+	update_icon()
```

Comment:
why is this needed every process tick? You have 3 specific things that can change the icon state, you should be calling update icon specifically where those change.

Reply:
fixed

Reply:
Don't self resolve please. Are you sure you've actually done this correctly as opposed to just removing this?

Reply:
well unless you know something i don't the only two things updating the icon should be update_icon_state and go_out, (plus the other checks like occupant and whatnot but those are conditional)

Reply:
I only see the icon being update in go_out. What about going in?

Reply:
> I only see the icon being update in go_out. What about going in?

that would be move_inside_wrapper which deals with the occupant var, which in turn is involved with update_icon_state

Reply:
? Unless I am blind, that doesn't call update icon

Reply:
![bruh](https://github.com/tgstation/TerraGov-Marine-Corps/assets/102577236/04bff3d3-2ad5-46b1-9737-490838fba3ea)

actually not even sure if it's even move_inside_wrapper since i didn't touch it
i just know the fix works and thats why i made the PR, if something is in fact broken then tell me

Reply:
Its not working. You removed it like I asked from processing but you didn't add it anywhere else it needs to be.

Reply:
it does not need to be anywhere
1. the code works, why would i bother putting that anywhere else
2. ![Capture](https://github.com/tgstation/TerraGov-Marine-Corps/assets/102577236/554d2fb8-f2f7-481e-9345-25400d8e0104) this handles the icon_change going out

Reply:
Bruh, I tested it, jump in, the icon doesn't update. Finishes the repair cycle and boots you out, it updates then.
So it never changes from the inactive state

Reply:
> Bruh, I tested it, jump in, the icon doesn't update. Finishes the repair cycle and boots you out, it updates then. So it never changes from the inactive state

guh, turns out synth can't even drag themself into a robotic cradle which is why i apparently didn't see this